app/controllers/spree/admin/product_catalogs_controller.rb

```diff
@@ -17,10 +18,13 @@ def collection
       end
 
       def load_data
-        @variants = Spree::Product.all.map(&:variants_including_master).flatten
         @product_catalogs = Spree::ProductCatalog.where(store: current_store).order(:name)
       end
 
+      def load_variants
+        @variants = Spree::Product.all.map(&:variants_including_master).compact.flatten
```

Comment:
This change is more computationally expensive, because it is guaranteed to instantiate every product and variant with no opportunity to use `find_each` or something similar.

Reply:
How are we using @variants in the view? Again, if we are building a <select> element then there are some good clean ways to optimize that using a lazily evaluated AR scope and a helper method.

Reply:
actually we just need to call `Spree::Variant.all`, I made the change here https://github.com/geminimvp/solidus_product_feed/pull/11/files#diff-154536b14bceed66fa7c2eec20e60704R25

Reply:
Again, we cannot iterate over `Spree::Variant.all` in any request context. That is memory and time-intensive.

Reply:
now I'm using this approach https://github.com/geminimvp/solidus_product_feed/pull/11/commits/630bf82b5f06b1ef578a65e1240ab9125aef3c6c#diff-154536b14bceed66fa7c2eec20e60704R25, please let me know what do you think